src/main/java/com/redislabs/ReadSource.java

```diff
@@ -123,7 +117,7 @@ public void foreach(KeysReaderRecord r) throws Exception {
         }
       }
       
-      if(response.size() > 0 && originalCmd.equals("hget")) {
+      if(response.isEmpty() && originalCmd.equals("hget")) {
```

Comment:
!response.isEmpty() no?

Reply:
Right!!! good catch...

Reply:
Its a bad catch, the tests should have caught it. At least it reminds me that I did not add tests for read through ...